### ISSUE TITLE
`cannot borrow `function_block` as mutable, as it is not declared as mutable` when trying to use library

### ISSUE THREAD

**Issue by qarmin** [stevenliebregt/fun_time#4]
```
    #[fun_time]
    fn find_empty_folders_internal(&mut self, stop_receiver: Option<&Receiver<()>>, progress_sender: Option<&UnboundedSender<ProgressData>>) {
        self.optimize_dirs_before_start();
```
prints error
```
error[E0596]: cannot borrow `function_block` as mutable, as it is not declared as mutable
  --> czkawka_core/src/empty_folder.rs:58:5
   |
58 |     #[fun_time]
   |     ^^^^^^^^^^^ cannot borrow as mutable
59 |     fn find_empty_folders_internal(&mut...
60 |         self.optimize_dirs_before_start();
   |         ---- calling `function_block` requires mutable binding due to mutable borrow of `*self`
   |
   = note: this error originates in the attribute macro `fun_time` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider changing this to be mutable
   |
58 |     mut #[fun_time]
   |     +++

```
suggestion of course, not works

**Comment by stevenliebregt** [comment]
Thanks for your issue, I never thought about struct members. It should be fixed now in version `0.3.1` https://github.com/stevenliebregt/fun_time/releases/tag/0.3.1

**Comment by stevenliebregt** [comment]
The fix for this introduced a regression, if your function has a `return` statement in it the time will never be logged. As it expands to:

```rust
let start_time = ..;

let return_value = {
    // This makes the entire function return, originally with wrapping the code in a function call
    // that was avoided.
    return my_original_fn_return;
}

let elapsed = start_time.elapsed(); // Can not be reached now
```

I will revert this and look into this again

**Comment by stevenliebregt** [comment]
Fixed again in `0.3.2` https://github.com/stevenliebregt/fun_time/releases/tag/0.3.2
